### PR TITLE
style: Enable `phpdoc_list_type` fixer

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Enables `phpdoc_list_type`, changing `array` phpdocs with implicit keys to `list`, e.g. `array<string> => list<string>`. Then let phpstan catch incorrect usages.

**If approved, don't merge yet so this can be applied in the coding standard repo and minimize updating the overrides here.**

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
@paulbalandan Sorry, I've changed to use `@property-read`. See #8519.
Can you update coding-standard with that?

**Comment by kenjis** [comment]
`phpdoc_list_type` is risky.
https://cs.symfony.com/doc/rules/phpdoc/phpdoc_list_type.html

I'm not sure we should enable this rule by default.

**Comment by paulbalandan** [comment]
Yes, it is risky by the reasoning behind it.

If we make a phpdoc like `Foo[]`, I think we almost always mean the same way when we push to an array, like `$a[] = Foo;`, so `Foo[]` here would mean a list of `Foo`, i.e., `array<int, Foo>`. If we mean a string key, we should have indicated it like `array<string, Foo>`, right?

File: app/Config/View.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ class View extends BaseView
      * by the core Parser by creating aliases that will be replaced with
      * any callable. Can be single or tag pair.
      *
-     * @var array<string, array<string>|callable|string>
+     * @var array<string, callable|list<string>|string>
      * @phpstan-var array<string, array<ParserCallableString>|ParserCallableString|ParserCallable>
```
Comment (kenjis) [inline app/Config/View.php:48]:
```suggestion
     * @phpstan-var array<string, list<ParserCallableString>|ParserCallableString|ParserCallable>
```

File: system/Config/View.php [hunk 2]
```diff
@@ -51,7 +51,7 @@ class View extends BaseConfig
      *
      * @psalm-suppress UndefinedDocblockClass
      *
-     * @var array<string, array<string>|callable|string>
+     * @var array<string, callable|list<string>|string>
      * @phpstan-var array<string, array<parser_callable_string>|parser_callable_string|parser_callable>
```
Comment (kenjis) [inline system/Config/View.php:55]:
```suggestion
     * @phpstan-var array<string, list<parser_callable_string>|parser_callable_string|parser_callable>
```

File: system/Config/View.php [hunk 3]
```diff
@@ -89,7 +89,7 @@ class View extends BaseConfig
     /**
      * Built-in View plugins.
      *
-     * @var array<string, array<string>|callable|string>
+     * @var array<string, callable|list<string>|string>
      * @phpstan-var array<string, array<parser_callable_string>|parser_callable_string|parser_callable>
```
Comment (kenjis) [inline system/Config/View.php:93]:
```suggestion
     * @phpstan-var array<string, list<parser_callable_string>|parser_callable_string|parser_callable>
```

File: app/Config/View.php (hunk 1 above)
Comment (kenjis) [reply app/Config/View.php:48]: Can you change the type names? These are not a classname, so `parser_callable` is better.
See `CodeIgniter\Config\View`.

```
 * @phpstan-type parser_callable (callable(mixed): mixed)
 * @phpstan-type parser_callable_string (callable(mixed): mixed)&string
```

File: system/View/Parser.php [hunk 4]
```diff
@@ -63,7 +63,7 @@ class Parser extends View
     /**
      * Stores any plugins registered at run-time.
      *
-     * @var array<string, array<string>|callable|string>
+     * @var array<string, callable|list<string>|string>
      * @phpstan-var array<string, array<parser_callable_string>|parser_callable_string|parser_callable>
```
Comment (kenjis) [inline system/View/Parser.php:67]:
```suggestion
     * @phpstan-var array<string, list<parser_callable_string>|parser_callable_string|parser_callable>
```

**Comment by kenjis** [comment]
> If we make a phpdoc like Foo[], I think we almost always mean the same way when we push to an array, like $a[] = Foo;, so Foo[] here would mean a list of Foo, i.e., array<int, Foo>. If we mean a string key, we should have indicated it like array<string, Foo>, right?

I checked all the changes in this PR, and you are right.
I agree that most PHP devs think like that.

I agree to enable `phpdoc_list_type`.

**Review by kenjis — APPROVED** [review]

**Comment by paulbalandan** [comment]
To be merged once latest coding-standard is out.

**Review by kenjis — APPROVED** [review]
The Psalm errors are not directly related to this PR.

**Pull request merged**